app/controllers/cashierController.js

```diff
@@ -0,0 +1,78 @@
+import accountModel from '../models/accountModel';
+import transactionModel from '../models/transactionModel';
+
+export default class CashierController {
+  static async creditAccount(req, res) {
+    try {
+      let oldBalance;
+      const cashier = req.user.userId;
+      const { amount } = req.body;
+      const createdOn = new Date();
+      const lastTransaction = await transactionModel.getLastTransaction(req);
+      if (lastTransaction) {
+        oldBalance = lastTransaction.accountbalance;
+      } else {
+        const bankAccount = await accountModel.getAccount(req, res);
+        oldBalance = bankAccount[0].openingbalance;
+      }
+      const accountBalance = ((+oldBalance) + (+amount)).toFixed(2);
+      const data = {
+        type: 'credit', amount, cashier, accountBalance, createdOn,
+      };
+      const newTransaction = await transactionModel.create(req, data);
+      if (newTransaction.length) {
+        return res.status(200).json({
+          status: 200,
+          message: `The account ${req.params.accountNumber} has been credited with ${amount} on ${createdOn}`,
+          data: newTransaction[0],
+        });
+      }
+      return null;
+    } catch (error) {
+      return res.status(500).send({
+        status: 500,
+        error: 'Unable to credit account!! Server Error, Please Try Again',
+      });
+    }
+  }
+
+  static async debitAccount(req, res) {
+    try {
+      let oldBalance;
+      const cashier = req.user.userId;
+      const { amount } = req.body;
+      const createdOn = new Date();
+      const lastTransaction = await transactionModel.getLastTransaction(req);
+      if (lastTransaction) {
+        oldBalance = lastTransaction.accountbalance;
+      } else {
+        const bankAccount = await accountModel.getAccount(req, res);
+        oldBalance = bankAccount[0].openingbalance;
+      }
+      if ((+oldBalance) < (+amount)) {
+        return res.status(409).json({
```

Comment:
409 is code for conflict. I think 400 is better for this

Reply:
ouch, will fix it